### PR TITLE
Add rolify gem to assign users roles for the application

### PR DESCRIPTION
**Pull request by cbeer**

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -10,12 +10,19 @@ class Ability
   def custom_permissions
     # Limits deleting objects to a the admin user
     #
-    can [:manage], :all if admin?
+    can [:manage], :all if current_user.has_role? :admin
```
Comment (mjgiarlo) [inline app/models/ability.rb]: I'm guessing you want to revert this change since you made an `#admin?` method below.

File: app/models/user.rb [hunk 2]
```diff
@@ -15,10 +16,18 @@ class User < ActiveRecord::Base
...
   # user class to get a user-displayable login/identifier for
   # the account.
   def to_s
     email
   end
+
+  private
+
+    def add_default_roles
+      add_role :admin unless self.class.any?
```
Comment (mjgiarlo) [inline app/models/user.rb]: What's the effect of `unless self.class.any?` here?

**Comment by mjgiarlo** [comment]
:+1: 

:eyes: Travis

File: app/models/ability.rb (hunk 1 above)
Comment (cbeer) [inline app/models/ability.rb]: Nope. I hate `Ability#admin?` and intend to remove it upstream.

Comment (mjgiarlo) [inline app/models/ability.rb]: Oh, is that private method below only to satisfy rolify? I saw the duplication in code and balked.

Comment (cbeer) [inline app/models/ability.rb]: No, to satisfy.. sufia? curation_concerms?

Comment (mjgiarlo) [inline app/models/ability.rb]: OK. np.
